test/clojure/nrepl/bencode_test.clj

```diff
@@ -196,3 +197,15 @@
                (decode :reader read-bencode)
                (get "data")
                seq)))))
+
+(deftest unwritable-values
+  (testing "write-bencode writes eagerly"
+    (let [out (ByteArrayOutputStream.)]
+      (is (thrown? IllegalArgumentException
+                   (write-bencode out {"obj" (Object.)})))
+      (is (= "d3:obj" (String. (.toByteArray out))))))
```

Comment:
I guess you can also use `byte-array`. It should behave the same way, right?

Reply:
Not sure if you can stream to a `byte-array`. Also seems the fixed size part might be a bad fit for what we need to do here?

Reply:
I meant Clojure has a `byte-array` function which I think does the same thing as `.toByteArray`. :-)

Reply:
Tried, and afraid it doesn't. ByteArrayOutputStream doesn't implement ISeq.